contributing.md

```diff
@@ -0,0 +1,257 @@
+
+### Commands to check before finishing a pull request!
+
+Every command of the CLI inside the project must be operative and the tests before your changes cannot fail on commit, if you implemented a new feature and this feature increments or decrements the operations, you need to update the test file and the [readme.md](./readme.md) or open an [issue](https://github.com/victorperin/qr-scanner-cli/issues/new) for changing the documentation with a great description for what to do and referencing the feature.
+
+### Husky the pre-commit flow
+
+Husky is an hook that operates before commit changes in the project the hook lint the commit message with commitlint to follow a guideline, on the development of this project you need to follow some those to can commit your project, the commit message has 
+
+#### Remember to follow it, otherwise your code will be rejected
+
+The commit message need keep a standard, so we chose to use the commitlint with angular configuration, the fully documentation can be [find here](https://www.npmjs.com/package/@commitlint/config-angular), but the abstract of the documentation will be bottom:
+
+The commitmessage need to start with a type, the acceptable type's is:
+```javascript
+types => ['build','ci','docs','feat','fix','perf','refactor','revert','style','test']
+```
+
+And the type-case is ruled by lowercase always if is in uppercase or camelcase is not valid.
+
+After the type is optional to describe the scope of changes also in lowercase to be valid and if is used need to be inside pharentesis other case if is not used need to be nothing has changed.
+
+And ```:``` with the message of commit objective; the full commit message wold'nt finish with ```.``` and can't pass to 72 chars or will be disconsidered by hook.
+
+> To keep pretty, please dedicate an space before the message commit
+
+These are some cases to exemplify the commit message:
+
+**Commit message**|**Accpetable**
+-----|-----
+TYPE:commit message|:-1:
+Type:commit message|:-1:
+type:commit message|:+1:
+:commit message|:-1:
+type: commit message is too long to this way i can exemplification everything and keep my history today|:-1:
+type: commit message.|:-1: 
+type: commit message|:+1:
+type(SCOPE): commit message|:-1:
+type(Scope): commit message|:-1:
+type(): commit message|:-1:
+type(scope): commit message|:+1:
+
+More examples you can find in the full documentation.
+
+## How Can I Contribute?
+
+### Reporting Bugs
+
+This section guides you through submitting a bug report for QR Scanner CLI. Following these guidelines helps maintainers and the community understand your report :pencil:, reproduce the behavior :man-technologist:, and find related reports :mag_right:.
```

Comment:
For some reason :man-technologist: doesn't worked on preview, i think github hasn't implemented.